thunder/torch/__init__.py

```diff
@@ -1812,6 +1812,14 @@ def leaky_relu(a: TensorProxy, /, negative_slope: float = 0.01, inplace: bool =
 _inplace_to_out_of_place[leaky_relu] = leaky_relu, 2
 
 
+@torchsymbol(torch.nn.functional.logsigmoid, is_method=False)
+def logsigmoid(a: TensorProxy, /):
+    return log(sigmoid(a))
```

Comment:
In SciPy, the function that Pytorch calls `logsigmoid` is called `log_expit`:

https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.log_expit.html

and the implementation is:

https://github.com/scipy/scipy/blob/b589120c24e45baf03aabeb7fcb5e68d5f2c4de3/scipy/special/xsf/log_exp.h#L60

In PyTorch the CUDA implementation is:

https://github.com/pytorch/pytorch/blob/12b8c2fd8b91fb54abb50b25f3618bd5150125aa/aten/src/ATen/native/cuda/ActivationLogSigmoidKernel.cu#L25

What do you think of those formulations, and what about PyTorch's custom logsigmoid backward?

https://github.com/pytorch/pytorch/blob/12b8c2fd8b91fb54abb50b25f3618bd5150125aa/aten/src/ATen/native/cuda/ActivationLogSigmoidKernel.cu#L43

If we implement logsigmoid like SciPy and PyTorch, would a custom logsigmoid backward reduce memory use or make forward->backward faster than the autogenerated backward?

Reply:
While it doesn't look like much, there was an awful lot of trial and error that went into this revision.  

I think I understand how the scipy version of the forward impl improves the efficiency and accuracy.  I'm not sure I understand how a custom impl for the backward version is a major improvement on what would be automatically generated from this the scipy-like forward impl, though, the trace does look "tighter".